packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,152 @@
+import { KEY_SPACE, KEY_RETURN } from 'keycode-js';
+import { enableFocusStyles, removeFocusStyles, restoreFocusStyles } from './_SwitchUtils';
+import styles from './Switch.module.scss';
+
+const cx = classNamesBind.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: <FormattedMessage id="Terra.switch.switchstatuslabel.on" />,
+  OFF: <FormattedMessage id="Terra.switch.switchstatuslabel.off" />,
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  switchId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  isChecked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * The label text of the Switch component.
+   */
+  labelText: PropTypes.string.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Returns Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isChecked: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    switchId,
+    isChecked,
+    isDisabled,
+    onChange,
+    labelText,
+    ...customProps
+  } = props;
+  const theme = React.useContext(ThemeContext);
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback((event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+    sliderButton.current.focus();
+    if (onChange) {
+      onChange(!isChecked, event);
+    }
+  }, [isChecked, onChange]);
+
+  const handleOnKeyDown = (event) => {
+    if (event.nativeEvent.keyCode === KEY_RETURN || event.nativeEvent.keyCode === KEY_SPACE) {
+      event.preventDefault();
+      enableFocusStyles(sliderButton.current);
+      if (onChange) {
+        onChange(!isChecked, event);
+      }
+    }
+  };
+
+  const handleOnMouseDown = (event) => {
+    event.preventDefault();
+    removeFocusStyles(sliderButton.current);
+  };
+
+  const handleOnBlur = () => {
+    restoreFocusStyles(sliderButton.current);
+  };
+
+  const labelClassNames = cx([
+    'label-text',
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const statusLabelClassNames = cx([
+    'status-label-text',
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { 'is-selected': isChecked },
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-selected': isChecked },
+    { 'is-disabled': isDisabled },
+  ]);
+
+  const statusLabelText = isChecked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  const switchClassNames = classNames(cx(
+    'switch',
+    { 'is-disabled': isDisabled },
+    theme.className,
+  ),
+  customProps.className);
+
+  delete customProps.className;
+  return (
+    <button
```

Comment:
Set button to div, this will resolve the focus issue with the associated css change.

Reply:
Are you referring to prevent default of mouseDown Event..?

Reply:
changed button to div : [22dedf6f](https://github.com/cerner/terra-core/pull/2846/commits/893ebe62f9fd9578c95be4985af8c5a9b9545525#diff-ce03daf8e4362de3a438b2f322dedf6f) . but we need preventDefault on both handleKeyDown and handleMouseDown to keep switch focus styles consistent in IE.